dashboard/serializers.py

```diff
@@ -38,6 +39,29 @@ def serialize_enrollments(cls, mmtrack, enrollments):
         return serialized_enrollments
 
     @classmethod
+    def count_courses_passed(cls, program, mmtrack):
+        """
+        Returns number of passed courses.
+
+        Args:
+            program (Program): A program object.
+            mmtrack (dashboard.utils.MMTrack): a instance of all user information about a program
+        Returns:
+            num_complete_courses (int): A number of passed courses.
+        """
+        num_courses_passed = 0
+        for course in program.course_set.all():
+            course_info_dict = get_info_for_course(course, mmtrack)
+            if course_info_dict:
+                expected_statuses = list(
+                    filter(lambda c: c['status'] in [CourseStatus.PASSED], course_info_dict["runs"])
+                )
+                if len(expected_statuses) > 0:
+                    num_courses_passed += 1
+
+        return num_courses_passed
```

Comment:
i also don't understand "do this refactoring in a separate PR, to keep this one small and focused on the issue". the changes i'm suggesting are _directly_ related to this issue, and is 4 lines of code shorter (if you add the docstring on `count_courses_passed` which i forgot in my gist)